subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/RepositoryDetails.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.internal.artifacts.repositories;
+
+import org.gradle.internal.scan.UsedByScanPlugin;
+
+import java.util.Map;
+
+/**
+ * Details about a repository.
+ * See {@link ResolutionAwareRepository} and {@link org.gradle.api.internal.artifacts.ivyservice.ivyresolve.ModuleComponentRepository}
+ */
+public class RepositoryDetails {
+
+    /**
+     * A _unique_ identifier for a repository.
+     */
+    public final String id;
+
+    public final String name;
+    public final RepositoryType type;
+    public final Map<RepositoryPropertyType, ?> properties;
+
+    public RepositoryDetails(String id, String name, RepositoryType type, Map<RepositoryPropertyType, ?> properties) {
+        this.id = id;
+        this.name = name;
+        this.type = type;
+        this.properties = properties;
+    }
+
+    @UsedByScanPlugin("doesn't link against this type, but expects these values - See ResolveConfigurationDependenciesBuildOperationType")
+    public enum RepositoryPropertyType {
+
+        URL,
+        DIRS,
+        ARTIFACT_URLS,
+        IVY_PATTERNS,
```

Comment:
Two questions:

1. Is the `URL` by itself sufficient to convey to the scans viewer what the repo really is? i.e. do these attributes convey something logical about the repository?
2. In practice do we see > 1 repo with the same URL but differing in these properties?

If it's 1:yes 2:no, then we probably aren't providing much value in conveying them to the user. 

It's not clear what your objection actually is here. I'm all for not having unnecessary things, but I'd like to know the reasoning.

If necessary, we should step back and do some design work to finalize what would actually be useful to 1) differentiate repositories, 2) communicate to users about repositories.

Reply:
I think in general a single `URL` will be sufficient to allow a user to differentiate between different repositories in the same build. While it's possible to have multiple repositories defined with the same base URL, I believe this would be very rare.

The remaining properties provide the full configured state of a repository. These values could be useful to determine _why_ a certain module was/wasn't found in a particular repository. I'm not sure that we are attempting to solve this use case, and if so I suspect that the actual locations searched for a module will provide a more direct insight.

So my general concerns are two-fold:

1. If we only require properties to differentiate between multiple repositories in a single build invocation, then the `name` +  base `URL` will be adequate in the vast majority of cases.

2. If we want to provide more details about the configuration of a particular repository and the things it can resolve, then certain proposed properties like `LAYOUT_NAME` and `IVY_PATTERNS` should be normalized into more fundamental properties. In other words, there are different configuration options that can produce the _same_ repository structure, and I'd think we'd want to normalize these to the _effective_ properties.